source/iostreams.tex

```diff
@@ -12227,10 +12227,10 @@
 \end{note}
 
 \pnum
-If the native format requires paths for regular files to be formatted
-differently from paths for directories, the path shall be treated as a directory
-path if its last element is a \grammarterm{directory-separator},
-otherwise it shall be treated as a path to a regular file.
+If the native format requires paths for directories to be formatted
+differently from paths for other types of files,
+the path shall be treated as a path to a directory if and only if
+its last element is a \grammarterm{directory-separator}.
```

Comment:
It's unclear whether the subject of "its last element" is the native format or the path. I assume it's the former because except for a _root-directory_, an element of a path is never a _directory-separator_. I think this should say "the native format's last element" or "the last character of the native format" instead.

Reply:
This is pre-existing.

I think "its" refers to the textual representation of the path before it's converted to the native format.

For example, for a hypothetical implementation on OpenVMS, it might be the case that

```
path("foo/bar/", path::format::generic_format).native() // returns "[foo.bar]"
path("foo/bar",  path::format::generic_format).native() // returns "[foo]bar"
```

Reply:
Actually, [old boost.filesystem documentation](https://www.boost.org/doc/libs/1_48_0/libs/filesystem/v3/doc/reference.html) says:

> If the native format requires regular file paths and directory paths to be formatted differently, the implementation shall determine which format to use according to whether or not the last element of the generic format string is a separator. [Example: On OpenVMS, a path constructed from `"/cats/jane"` would considered a regular file path, and have a native format of `"[CATS]JANE"`, while a path constructed from `"/cats/jane/"` would be considered a directory path, and have a native format of `"[CATS.JANE]"`. --end example] [Note: POSIX and Windows use the same native format for both regular file and directory pathnames, so this paragraph does not apply to them. --end note]

Reply:
I think "last character of the generic format string" would be correct and unambiguous.

Reply:
done